src/Functions/registerFunctions.cpp

```diff
@@ -3,143 +3,40 @@
-    registerFunctionsArray(factory);
-    registerFunctionsTuple(factory);
-    registerFunctionsMakeDate(factory);
-    registerFunctionsMap(factory);
-    registerFunctionsBitmap(factory);
-    registerFunctionsBinaryRepr(factory);
-    registerFunctionsCoding(factory);
-    registerFunctionsCodingUUID(factory);
-    registerFunctionChar(factory);
-    registerFunctionsComparison(factory);
-    registerFunctionsConditional(factory);
-    registerFunctionsConversion(factory);
-    registerFunctionCastOrDefault(factory);
-    registerFunctionsDateTime(factory);
-    registerFunctionsEmbeddedDictionaries(factory);
-    registerFunctionsExternalDictionaries(factory);
-    registerFunctionsExternalModels(factory);
-    registerFunctionsFormatting(factory);
-    registerFunctionsHashing(factory);
-    registerFunctionsHigherOrder(factory);
+    // These two functions are not build with clickhouse_function,
+    // need to be registered separately
     registerFunctionsLogical(factory);
-    registerFunctionsMiscellaneous(factory);
-    registerFunctionsRandom(factory);
-    registerFunctionsReinterpret(factory);
-    registerFunctionsRound(factory);
-    registerFunctionsString(factory);
-    registerFunctionsStringArray(factory);
-    registerFunctionsStringSearch(factory);
-    registerFunctionsStringRegexp(factory);
-    registerFunctionsStringSimilarity(factory);
-    registerFunctionsStringTokenExtractor(factory);
-    registerFunctionsURL(factory);
-    registerFunctionsVisitParam(factory);
-    registerFunctionsMath(factory);
-    registerFunctionsGeo(factory);
-    registerFunctionsNull(factory);
-    registerFunctionsJSON(factory);
-    registerFunctionsSQLJSON(factory);
-    registerFunctionToJSONString(factory);
-    registerFunctionsIntrospection(factory);
-    registerFunctionsConsistentHashing(factory);
-    registerFunctionsUnixTimestamp64(factory);
-    registerFunctionBitHammingDistance(factory);
-    registerFunctionTupleHammingDistance(factory);
-    registerFunctionsStringHash(factory);
-    registerFunctionValidateNestedArraySizes(factory);
-    registerFunctionsSnowflake(factory);
-    registerFunctionsTimeWindow(factory);
-    registerFunctionToBool(factory);
-    registerFunctionMinSampleSize(factory);
-
-    //meilisearch
-    registerFunctionMeiliMatch(factory);
-
-#if USE_SSL
-    registerFunctionEncrypt(factory);
-    registerFunctionDecrypt(factory);
-    registerFunctionAESEncryptMysql(factory);
-    registerFunctionAESDecryptMysql(factory);
-    registerFunctionShowCertificate(factory);
-#endif
-    registerFunctionTid(factory);
-    registerFunctionLogTrace(factory);
-    registerFunctionHashID(factory);
+    registerFunctionReplicate(factory);
+
+#if !defined(__APPLE__)
+    for (func_entry *p = &__start_func_list; p < &__stop_func_list; ++p)
+        (*p)(factory);
+#else /* !__APPLE__ */
+    size_t sz = 0;
+    func_entry *p = reinterpret_cast<func_entry *>(getsectdata("__RODATA", "func_list", &sz) +
```

Comment:
Will this work in shared build?

Reply:
I believe there will be no much difference. Previously the linking happens when libclickhouse_functionsd.a is created, now happens when the final binary is created.

Reply:
But linking the final binary is heavier compared to `libclickhouse_functionsd.so`. It's even better to build shared objects instead of objects when shared build is enabled.